sources/packages/backend/apps/workers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -286,20 +286,29 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
   }
 
   /**
-   * Updates assessment status and save workflow data.
+   * Updates assessment status and save workflow data ensuring
+   * that the data will be updated only once.
    * @param assessmentId updated assessment.
    * @param workflowData workflow data to be saved.
+   * @param entityManager used to execute the commands in the same transaction.
    */
   async updateAssessmentStatusAndSaveWorkflowData(
     assessmentId: number,
     workflowData: WorkflowData,
-  ): Promise<void> {
+    entityManager: EntityManager,
+  ): Promise<boolean> {
+    const studentAssessmentRepo =
+      entityManager.getRepository(StudentAssessment);
     const auditUser = this.systemUsersService.systemUser;
     const now = new Date();
-    const studentAssessment = await this.repo.findOne({
-      select: { studentAssessmentStatus: true },
+    const studentAssessment = await studentAssessmentRepo.findOne({
+      select: { studentAssessmentStatus: true, workflowData: true as unknown },
       where: { id: assessmentId },
     });
+    if (studentAssessment.workflowData) {
```

Comment:
minor. Should we also try to include the status ? 

```ts
        StudentAssessmentStatus.CancellationRequested,
        StudentAssessmentStatus.CancellationQueued,
        StudentAssessmentStatus.Cancelled,
       StudentAssessmentStatus.Completed,
```

Reply:
I was also in doubt here and I ended up checking only the `workflowData` since it would ensure that the method was already executed. For instance, if in the future we start saving more data I am not sure if we should also check this extra data.